stl/inc/ratio

```diff
@@ -19,59 +20,58 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
-template <intmax_t _Val>
-struct _Abs : integral_constant<intmax_t, (_Val < 0 ? -_Val : _Val)> {}; // computes absolute value of _Val
-
-template <intmax_t _Ax, intmax_t _Bx, bool _Sfinae = false,
-    bool _Good = (_Abs<_Ax>::value <= INTMAX_MAX / (_Bx == 0 ? 1 : _Abs<_Bx>::value))>
-struct _Safe_mult : integral_constant<intmax_t, _Ax * _Bx> {}; // computes _Ax * _Bx without overflow
+_NODISCARD constexpr intmax_t _Abs(const intmax_t _Val) noexcept { // computes absolute value sof _Val
+    return _Val < 0 ? -_Val : _Val; // caller ensures that _Val does not go below -INTMAX_MAX
+}
 
-template <intmax_t _Ax, intmax_t _Bx, bool _Sfinae>
-struct _Safe_mult<_Ax, _Bx, _Sfinae, false> { // _Ax * _Bx would overflow
-    static_assert(_Sfinae, "integer arithmetic overflow");
-};
+_NODISCARD constexpr intmax_t _Safe_mult(
+    const intmax_t _Ax, const intmax_t _Bx) noexcept { // computes _Ax * _Bx without overflow

```

Comment:
We try to avoid line breaks due to comments
```suggestion
_NODISCARD constexpr intmax_t _Safe_mult(const intmax_t _Ax, const intmax_t _Bx) noexcept { 
    // computes _Ax * _Bx without overflow
```